wp-cache-sanitizer.php

```diff
@@ -0,0 +1,726 @@
+
+    if ( ! in_array( $setting, $this->allowed_values ) )
+        return false;
+
+    if ( in_array( $setting, $this->allowed_booleans ) )
+      $sanitized_value = $this->sanitize_boolean( $setting, $value );
+    else if ( in_array( $setting, $this->allowed_integers ) )
+      $sanitized_value = $this->sanitize_int( $setting, $value );
+    else if ( in_array( $setting, $this->allowed_strings ) )
+      $sanitized_value = $this->sanitize_string( $setting, $value );
+    else if ( in_array( $setting, $this->allowed_options ) )
+      $sanitized_value = $this->sanitize_options( $setting, $value );
+
+    if ( $insert === true )
+        $this->insert( $setting, $sanitized_value );
+
+    return $sanitized_value;
+  }
+
+  /**
+   * This function inserts the current setting and its sanitized value into the wp-cache-config.php
+   * file or the database depending on the setting.
+   *
+   * @param String $setting The current setting being inserted
+   * @param Mixed $sanitized_value The sanitized value of the setting
+   *
+   * @since  1.4.4
+   *
+   * @uses  wp_cache_replace_line Replace the lines in the wp-cache-config.php file
+   */
+  private function insert( $setting, $sanitized_value ) {
+    global $wp_cache_config_file;
+
+    if ( in_array( $setting, $this->allowed_booleans ) ) {
+        wp_cache_replace_line('^ *\$' . $setting . ' =', "\$" . $setting . " = " . $sanitized_value . ";", $wp_cache_config_file);
+    } else if ( in_array( $setting, $this->allowed_strings ) ) {
+        if ( is_array( $sanitized_value ) && $setting === 'wp_cache_pages' ) {
+            foreach ( $sanitized_value as $page => $status ) {
+                wp_cache_replace_line('^ *\$wp_cache_pages\[ "' . $page . '" \]', "\$wp_cache_pages[ \"{$page}\" ] = $status;", $wp_cache_config_file);
+            }
+        } else {
+            wp_cache_replace_line('^ *\$' . $setting . ' =', "\$". $setting ." = $sanitized_value;", $wp_cache_config_file);
+        }
+    } else if ( in_array( $setting, $this->allowed_integers ) ) {
+        wp_cache_replace_line('^ *\$' . $setting . ' =', "\$". $setting ." = $sanitized_value;", $wp_cache_config_file);
+    } else if ( in_array( $setting, $this->allowed_options ) ) {
+        update_option( $setting, $sanitized_value );
+    }
+  }
+
+  /**
+   * This sanitizes the settings where the allowed values are boolean, 0 or 1. The boolean values true and false
+   * are converted to strings to maintain the values in wp-cache-config.php file
+   *
+   * @param String $setting The current boolean setting to be sanitized
+   * @param Mixed $value The mixed value of the current setting that will be sanitized
+   *
+   * @return Boolean|String A sanitized version of the inputed value
+   */
+  function sanitize_boolean( $setting, $value ) {
+    // Note: necessary to force cache_enable and super_cache_enabled to be true/false
+    $sanitized_value = apply_filters( "wp_super_cache_sanitize_$value", $value );
```

Comment:
The wp_super_cache_sanitize_ filter has been replaced with using the setting name instead of the value c8e65e36